site/content/blog/2019-12-20-hermes-js-engines.md

```diff
@@ -0,0 +1,26 @@
+---
+title: On Hermes and Mattermost
+slug: on-hermes-and-mattermost
+date: 2019-12-20T12:00:00-04:00
+author: Miguel Alatzar
+github: migbot
+community: miguel.alatzar
+---
+
+With the upgrade to [React Native 61](https://facebook.github.io/react-native/blog/2019/09/18/version-0.61) came the prospect of substantially improving performance of our Android app. How? Through the use of [Hermes](https://github.com/facebook/hermes), Facebook’s new JavaScript engine. To say that we were excited is an understatement. And with that excitement came curiosity: how is this new JavaScript engine achieving performance boosts? 
+
+Let’s first chat a bit about JS engines in general. 
+
+How does your JavaScript code eventually get executed by the CPU on the machine? Through a JavaScript engine. Initially JS engines interpreted JS code, that is, as each line came in during run-time the JS engine would translate it to machine code that the host machine then executed. Later, with the use of Just-In-Time (JIT) compilation, optimizations could be made as “hot code” can be compiled and stored so that it, instead of the source code, could be executed the next time it’s needed. 
```

Comment:
```suggestion
How does your JavaScript code eventually get executed by the CPU on the machine? Through a JavaScript engine. Initially JS engines interpreted the JS code, that is, as each line came in during run-time, the JS engine would translate it to a machine code that the host machine then executed. Later, with the use of Just-In-Time (JIT) compilation, optimizations could be made as “hot code” and could be compiled and stored so that it, instead of the source code, could be executed the next time it’s needed.
```

Reply:
Sounds good.